[Term Entry] Dart List - Added list.md which describes Dart Lists

Only-Codes opened the pull request:
<!--
👋 Hi, thanks for submitting a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

<!-- Please write a summary of the change, such as which topic(s) and file(s) that you have edited or created. Please also include relevant motivation and context: -->
This closes issue #4235. 

I have created a new file 'list.md' which provides information on Dart Lists, this includes a description of what lists are, examples of lists and some common ways to make changes to lists. 

### Type of Change

<!--- Please delete or cross off the bullet point(s) that are irrelevant to this PR: -->

- Adding a new entry

### Checklist

<!-- Please check ALL the boxes: -->

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Codecademy/docs?pullRequest=4282) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Codecademy/docs?pullRequest=4282) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Codecademy/docs?pullRequest=4282) it.</sub>

ishg-153 commented on content/dart/concepts/list/list.md:
```diff
@@ -0,0 +1,135 @@
+
+## Different List Types
+
+There are two types of lists in Dart; fixed length or growable. A fixed length list has a specified size which cannot be changed after initialistation, whereas a growable list can have data added or removed.
```
```suggestion
There are two types of lists in Dart, fixed-length and growable. A fixed-length list has a specified size that cannot be changed after initialization. When creating a fixed-length list, it starts empty and needs items to be added to it. It is useful when the number of items in a list is known in advance. The following code block demonstrates the initialization of a fixed-length list:
```

ishg-153 commented on content/dart/concepts/list/list.md:
````diff
@@ -0,0 +1,135 @@
+// Example of a growable list
+List growableList = ['Lion', 'Tiger', 'Leopard'];
+```
+
+When creating a growable list, we are able to start with a set of data. We can then add additional data or remove data without causing an error. Growable lists provide flexibility and grow or shrink dynamically based on requirements.
````
```suggestion
```

ishg-153 commented on content/dart/concepts/list/list.md:
````diff
@@ -0,0 +1,135 @@
+// Example of a fixed length list
+List fixedLengthList = List(3);
+```
+
````
```suggestion

>**Note:** If more than the specified number of items will result in an error.

```

ishg-153 reviewed (changes requested):
@Only-Codes I have suggested a couple of changes, please make those. With those also address the following changes:
1. Remove the use of all personal pronouns like you, we, etc.
2. Your entry solely focuses on the List class, also include the more prominently used method of declaring lists ` var thisIsAList = [1, 2, 3]`

Only-Codes commented:
> @Only-Codes I have suggested a couple of changes, please make those. With those also address the following changes:
> 
> 1. Remove the use of all personal pronouns like you, we, etc.
> 2. Your entry solely focuses on the List class, also include the more prominently used method of declaring lists ` var thisIsAList = [1, 2, 3]`

Hi ishg-153 - Thank you for the feedback! I'll get these amends made as soon as possible.

Only-Codes commented:
Hi @ishg-153 - I have now updated the document with your suggestions. I have also removed personal pronouns and updated the code examples to use 'var' instead of 'Lists' whilst keeping an example of 'Lists' for demonstration purposes.

ishg-153 reviewed (approved):
@Only-Codes Great! This looks good for a second review.

avdhoottt commented on content/dart/concepts/list/list.md:
````diff
@@ -0,0 +1,155 @@
+];
+```
+
+Then, an example of utilizing `List` to create a list:
````
```suggestion
Then, an example of utilizing `List` to create a list:

```

avdhoottt commented on content/dart/concepts/list/list.md:
```diff
@@ -0,0 +1,155 @@
+
+>**Note:** As `var` is the predominantly used method to create lists in Dart, the rest of the examples will use this moving forward.
+
+In the above examples, there are three separate lists:
+- A list of strings
+- A list of integers
+- A nested list
```
You can add this before the example b=code block. It would be better for the users to understand the code block.

avdhoottt commented on content/dart/concepts/list/list.md:
````diff
@@ -0,0 +1,155 @@
+var fixedLengthList = List(3);
+```
+
+>**Note:** If more than the specified number of items are added, it will result in an error.
````
```suggestion
> **Note:** If more than the specified number of items are added, it will result in an error.
```

avdhoottt commented on content/dart/concepts/list/list.md:
```diff
@@ -0,0 +1,155 @@
+>**Note:** If more than the specified number of items are added, it will result in an error.
+
+When creating a fixed-length list, it starts empty and needs items added to it. Fixed-length lists are useful when a known limit on list items is required ahead of time. This enforces the list to stay at the specified limit.
+
```
Add a line describing the growable list example.

avdhoottt commented on content/dart/concepts/list/list.md:
````diff
@@ -0,0 +1,155 @@
+
+ages[0] = 87; // Adds 87 to the first position of the list
+```
+This comes in handy with fixed length lists:
````
```suggestion

This comes in handy with fixed-length lists:
```
Give one line space before and after a code block.

avdhoottt commented on content/dart/concepts/list/list.md:
````diff
@@ -0,0 +1,155 @@
+```
+
+## Displaying List Data
+To display data held within a list the `print` method is used. This can be used to display the whole list, or specific items within the list.
````
```suggestion

To display data held within a list the `print` method is used. This can be used to display the whole list, or specific items within the list.
```

avdhoottt commented on content/dart/concepts/list/list.md:
````diff
@@ -0,0 +1,155 @@
+To display data held within a list the `print` method is used. This can be used to display the whole list, or specific items within the list.
+
+```dart
+// Displaying a whole list
+var flowers = ['Rose', 'Lily', 'Sunflower', 'Tulip', 'Daisy'];
+
+print(flowers); // Displays ['Rose', 'Lily', 'Sunflower', 'Tulip', 'Daisy'] in the console.
+
+// Displaying a specific item from a list
+var musicalInstruments = ['Piano', 'Guitar', 'Violin', 'Trumpet'];
+
+print(musicalInstruments[2]); // Displays Violin in the console
````
Wrap this code under the main() function so that it will be executable as it is.

avdhoottt reviewed (changes requested):
Hey @Only-Codes, excellent entry. I have suggested some changes, please make them.

avdhoottt commented on content/dart/concepts/list/list.md:
````diff
@@ -0,0 +1,166 @@
+To display data held within a list the `print` method is used. This can be used to display the whole list, or specific items within the list.
+
+```dart
+/void main() {
````
```suggestion
void main() {
```

avdhoottt reviewed (approved):
LGTM!!

ishg-153 commented:
@avdhoottt should I merge this?

avdhoottt commented:
> @avdhoottt should I merge this?

Yup please go ahead. Thanks!!
